src/snowflake/connector/auth/by_plugin.py

```diff
@@ -197,6 +180,7 @@ def handle_timeout(
         account: str,
         user: str,
         password: str,
+        delete_params: bool = True,
```

Comment:
Because subclasses of `AuthByPlugin` (in my case `AuthByKeyPair`) may not want to delete `authenticator, service_name, account, user, password` while handling timeout. I added this so I could call `super().handle_timeout` in `AuthByKeyPair.handle_timeout` while preserving the original function's behaviour.